packages/angular/projects/ui-angular/src/lib/services/__tests__/authenticator.service.test.ts

```diff
@@ -0,0 +1,60 @@
+import { AuthenticatorService } from '../authenticator.service';
+
+import * as UIModule from '@aws-amplify/ui';
+import * as XState from 'xstate';
+
+// mock state machine service
+// based on https://github.com/statelyai/xstate/blob/main/packages/core/src/interpreter.ts
+export class MockAuthService {
```

Comment:
Unlike vue/angular, Angular uses `xstate` directly without any util library (`@xstate/react`), so had to do mocks at xstate level.

Reply:
👍